docs/src/locators.md

```diff
@@ -304,61 +285,86 @@ page.get_by_placeholder("name@example.com").fill("playwright@microsoft.com")
+```
+
+You can locate the element by the text it contains:
 
 ```js
-await page.getByText('Log in').click();
-await page.getByText('Log in', { exact: true }).click();
-await page.getByText(/log in$/i).click();
+await expect(page.getByText('Welcome, John')).toBeVisible();
+await expect(page.getByText('Welcome, John', { exact: true })).toBeVisible();
+await expect(page.getByText(/welcome, john$/i)).toBeVisible();
 ```
+
 ```java
-page.getByText("Log in").click();
-page.getByText("Log in", new Page.GetByTextOptions().setExact(true)).click();
-page.getByText(Pattern.compile("log in$", Pattern.CASE_INSENSITIVE)).click();
+assertThat(page.getByText("Welcome, John")).isVisible();
+assertThat(page.getByText("Welcome, John", new Page.GetByTextOptions().setExact(true))).isVisible();
+assertThat(page.getByText(Pattern.compile("welcome john$", Pattern.CASE_INSENSITIVE))).isVisible();
 ```
+
 ```python async
-await page.get_by_text("Log in").click()
-await page.get_by_text("Log in", exact=True).click()
-await page.get_by_text(re.compile("Log in", re.IGNORECASE)).click()
+await expect(page.get_by_text("Welcome, John")).to_be_visible()
+await expect(page.get_by_text("Welcome, John", exact=True)).to_be_visible()
+await expect(page.get_by_text(re.compile("welcome john", re.IGNORECASE))).to_be_visible()
 ```
+
 ```python sync
-page.get_by_text("Log in").click()
-page.get_by_text("Log in", exact=True).click()
-page.get_by_text(re.compile("Log in", re.IGNORECASE)).click()
+expect(page.get_by_text("Welcome, John")).to_be_visible()
+expect(page.get_by_text("Welcome, John", exact=True)).to_be_visible()
+expect(page.get_by_text(re.compile("welcome john", re.IGNORECASE))).to_be_visible()
 ```
+
 ```csharp
-await page.GetByText("Log in").ClickAsync();
-await page.GetByText("Log in", new() { Exact: true }).ClickAsync();
-await page.GetByText(new Regex("Log in", RegexOptions.IgnoreCase)).ClickAsync();
+await Expect(page.GetByText("Welcome, John")).ToBeVisibleAsync();
+await Expect(page.GetByText("Welcome, John", new() { Exact: true })).ToBeVisibleAsync();
+await Expect(page.GetByText(new Regex("welcome john", RegexOptions.IgnoreCase))).ToBeVisibleAsync();
 ```
 
-You can also [filter by text](#filter-by-text) when locating in some other way, for example find a particular item in the list.
+You can also [filter by text](#filter-by-text) which can be useful when trying to find a particular item in a list.
 
 ```js
-await page.getByTestId('product-item').filter({ hasText: 'Playwright Book' }).click();
+await page.getByTestId('product-item')
+    .filter({ hasText: 'Playwright Book' }).click();
 ```
 ```java
-page.getByTestId("product-item").filter(new Locator.FilterOptions().setHasText("Playwright Book")).click();
+page.getByTestId("product-item")
+    .filter(new Locator.FilterOptions().setHasText("Playwright Book")).click();
 ```
 ```python async
-await page.get_by_test_id("product-item").filter(has_text="Playwright Book").click()
+await page.get_by_test_id("product-item")
```

Comment:
This is not valid python, it must either be a single line, or use a continuation character:

```py
await page.get_by_test_id("product-item") \
    .filter(has_text="Playwright Book").click()
```

I think single line is more user-friendly in this case.